Wl 3776

surya-janani opened the pull request:
The logic in the resources view, greying out of resources is based on the selected parent's hidden attribute value.

In the collectionAccessFormatter , it was previously checking hidden attribute value for each of the child items. I've moved the line setting css, above the for loop iteration of the members.  So it now sets inactive based on root and not on each of its members.

For the scenario where there is a folder which is not hidden but which has one of its child item visible, I've again checked the visibility inside the for loop and set the visibility only if the child items visibility is explicitly set to hidden.

The alternative to the above solution would be to write a recursive method( in ResourceAction.java) that iterates through the tree structure of a particular selected source folder and set each of its child item's hidden attribute to true everytime something is hidden.

surya-janani committed:
fixing the greying out of resources by checking only the root element.-WL-3776

surya-janani committed:
Checking hidden value at the parent level -WL-3776

buckett commented on kernel-impl/src/main/java/org/sakaiproject/content/impl/CollectionAccessFormatter.java:
```diff
@@ -214,8 +214,10 @@ public void format(ContentCollection x, Reference ref, HttpServletRequest req, H
 					URI contentUri = new URI(contentUrl);
 					URI relativeUri = baseUri.relativize(contentUri);
 					contentUrl = relativeUri.toString();
+					if(content.isHidden()) {
```
I'd call `isAvailable()` again, otherwise if you're in a folder that's not hidden while iterating though the contents you come across a folder that is time restricted it won't get greyed out. `isAvailable()` is a little slower and does more checks that we know to be useless but I think it's sensible.

Otherwise you end up that when your view a time limited folder it will be greyed out, but when you see it in the parent folder listing it won't be.

I know time limiting was outside the original Jira but I think it's sensible to make the fix consistent, rather than introducing another edge case.

surya-janani committed:
WL-3776.
Resetting hidenclass at begininng of each iteration and using isAvailable()
method to make it work for timed resources.

surya-janani committed:
Moving isAvailable out of for loop
